service/src/test/java/uk/gov/hmcts/reform/fpl/service/RespondentServiceTest.java

```diff
@@ -54,6 +56,15 @@ void shouldBuildExpectedLabelWhenEmptyList() {
             assertThat(result).isEqualTo("No respondents on the case");
         }
 
+        @Test
+        void shouldSetPersistRepresentativeFlagToYes() {
+            List<Element<Respondent>> updatedRespondents = service.setPersistRepresentativeFlag(getRespondents());
+            List<Respondent> respondentsList = unwrapElements(updatedRespondents);
+
+            assertThat(respondentsList.get(0).getPersistRepresentedBy()).isEqualTo(YesNo.YES.getValue());
+            assertThat(respondentsList.get(1).getPersistRepresentedBy()).isEqualTo(YesNo.YES.getValue());
```

Comment:
As mentioned earlier the persistRepresentedBy could be YesNo type so getValue could be avoided.